### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by frederikja163**
Creates 27 basic unit tests that should cover the project nicely for a good start.
More unit tests can definitely and should be added in the future, but this serves as a good start.

### CODE REVIEW

File: Parquet/ResultListener/ParquetResultListener.cs [hunk 1]
```diff
@@ -133,15 +129,24 @@ private ParquetFile GetOrCreateParquetFile(SchemaBuilder schema, string path)
...
                 _parquetFiles[path] = file;
             }
-            else if (!file.CanContain(schema))
+            
+            if (!file.CanContain(builder.ToSchema()))
             {
+                builder.Union(file.Schema);
                 file.Dispose();
-                file = new ParquetFile(schema, path);
+                string tmpPath = path + ".tmp";
```
Comment (db-ks) [inline Parquet/ResultListener/ParquetResultListener.cs:140]: What happens if you already have a lingering .tmp file?

**Review by db-ks — APPROVED** [review]
Few comments, one inner PR, one commit, otherwise all looks good.

File: Parquet/ResultListener/ParquetResultListener.cs (hunk 1 above)
Comment (frederikja163) [reply Parquet/ResultListener/ParquetResultListener.cs:140]: It should get overriden, which is what we want as the file has a .tmp tag we shouldn't keep it imo.

File: Parquet.Tests/SchemaBuilderTests.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    {
+        Zero, One, Two
+    };
+
+    public class ParameterTestRun : TestRun
+    {
+
+        public ParameterTestRun()
+        {
+            Parameters.Add(new ResultParameter("Integer", 1));
```
[the remark this answers is not in the document]
Comment (frederikja163) [reply Parquet.Tests/SchemaBuilderTests.cs:18]: I think for now it makes more sense to add those tests if we experience errors. Although i will open an issue to add more thorough unit tests in general.

**Pull request merged**